### ISSUE TITLE
Navidrome scrobbling broken - auth works

### ISSUE THREAD

**Issue by bbsixzz** [krateng/maloja#186]
Navidrome 0.49.0 (25374b3b)

![Screen Shot 2023-02-03 at 06 51 28](https://user-images.githubusercontent.com/47334163/216523113-731d7b5c-2ebe-440d-91f8-dbb1e33632bb.png)

Maloja version: 3.1.4

```
2023/02/03 05:11:35  Listenbrainz API request: ['validate-token']
2023/02/03 05:11:53  Listenbrainz API request: ['submit-listens']
2023/02/03 05:11:53  Error with Listenbrainz API: <class 'maloja.apis._exceptions.InvalidAuthException'> (Request: [])
```

**Comment by bbsixzz** [comment]
Ok one step forward, the API key had no identifier so I made a new one but now I get this:

```
[apis] Listenbrainz API request: ['submit-listens']
[database] Incoming scrobble [Client: bbsixzz| API: Listenbrainz]: {'track_artists': ['Aphex Twin'], 'track_title': 'Acrid Avid Jam Shred', 'album_name': '...I Care Because You Do', 'scrobble_time': 1675405170, 'track_length': None, 'track_mbid': '002bb3b2-6986-4ac5-a7db-44724fd7ea68', 'release_mbid': 'e25e89cd-e505-4e5d-b2d3-733759750b47', 'artist_mbids': ['f22942a1-6f70-4f48-866e-238cb2308fbd']}
[apis] Error with Listenbrainz API: <class 'maloja.apis._exceptions.ScrobblingException'> (Request: [])
```

**Comment by bbsixzz** [comment]
If I try to scrobble manually:

```
[database] Incoming scrobble [Client: browser | API: native/v1]: {'track_artists': ['Aphex Twin'], 'track_title': 'Windowlicker'}
Traceback (most recent call last):
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/dbcache.py", line 47, in outer_func
    return cache[key]
KeyError: ('[{"artists": ["Aphex Twin"], "title": "Windowlicker", "album": {"name": null, "artists": null}, "length": null}]', '{}', <function connection_provider.<locals>.wrapper at 0x7fb4fccc7700>, None, None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/dbcache.py", line 47, in outer_func
    return cache[key]
KeyError: ('["Aphex Twin"]', '{}', <function connection_provider.<locals>.wrapper at 0x7fb4fccc78b0>, None, None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/apis/native_v1.py", line 89, in protector
    return func(*args,**kwargs)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/apis/native_v1.py", line 476, in post_scrobble
    result = database.incoming_scrobble(
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/__init__.py", line 97, in incoming_scrobble
    sqldb.add_scrobble(scrobbledict,dbconn=dbconn)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 120, in wrapper
    return func(*args,**kwargs)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 248, in add_scrobble
    add_scrobbles([scrobbledict],dbconn=dbconn)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 115, in wrapper
    return func(*args,**kwargs)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 255, in add_scrobbles
    ops = [
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 257, in <listcomp>
    **scrobble_dict_to_db(s,dbconn=dbconn)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 221, in scrobble_dict_to_db
    "track_id":get_track_id(info.get('track'),dbconn=dbconn),
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/dbcache.py", line 49, in outer_func
    result = inner_func(*args,**kwargs,dbconn=conn)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 115, in wrapper
    return func(*args,**kwargs)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 294, in get_track_id
    artist_ids = [get_artist_id(a,dbconn=dbconn) for a in trackdict['artists']]
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 294, in <listcomp>
    artist_ids = [get_artist_id(a,dbconn=dbconn) for a in trackdict['artists']]
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/dbcache.py", line 49, in outer_func
    result = inner_func(*args,**kwargs,dbconn=conn)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 115, in wrapper
    return func(*args,**kwargs)
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/maloja/database/sqldb.py", line 346, in get_artist_id
    op = DB['artists'].select(
TypeError: select() takes 1 positional argument but 2 were given
```

**Comment by bbsixzz** [comment]
`maloja info` gave me this:

```
Traceback (most recent call last):
  File "/usr/local/bin/maloja", line 8, in <module>
    sys.exit(main())
  File "/home/bbsixzz/.local/lib/python3.9/site-packages/doreah/control.py", line 51, in wrapper
    result = func(*args,**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/maloja/__main__.py", line 178, in main
    return action(*args,**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/maloja/__main__.py", line 140, in print_info
    print(col['lightblue']("Network:                "),f"IPv{ip_address(conf.malojaconfig['host']).version}, Port {conf.malojaconfig['port']}")
  File "/usr/lib/python3.9/ipaddress.py", line 53, in ip_address
    raise ValueError('%r does not appear to be an IPv4 or IPv6 address' %
ValueError: '*' does not appear to be an IPv4 or IPv6 address
```
I could access the web ui without setting the IP.

I gave it a proper IP now and `maloja info` shows no more errors, the original problem persists however.

**Comment by bbsixzz** [comment]
Linux debian 5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux

**Comment by bbsixzz** [comment]
I removed everything and built from source.

If I disable `Use global DB Cache` I remove this error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/dbcache.py", line 47, in outer_func
    return cache[key]
KeyError: ('["aaaa"]', '{}', <function connection_provider.<locals>.wrapper at 0x7f4bdcebd4c0>, None, None)
```

The rest still remains:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/maloja/apis/native_v1.py", line 97, in protector
    return func(*args,**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/maloja/apis/native_v1.py", line 484, in post_scrobble
    result = database.incoming_scrobble(
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/__init__.py", line 97, in incoming_scrobble
    sqldb.add_scrobble(scrobbledict,dbconn=dbconn)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 120, in wrapper
    return func(*args,**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 248, in add_scrobble
    add_scrobbles([scrobbledict],dbconn=dbconn)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 115, in wrapper
    return func(*args,**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 255, in add_scrobbles
    ops = [
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 257, in <listcomp>
    **scrobble_dict_to_db(s,dbconn=dbconn)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 221, in scrobble_dict_to_db
    "track_id":get_track_id(info.get('track'),dbconn=dbconn),
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/dbcache.py", line 49, in outer_func
    result = inner_func(*args,**kwargs,dbconn=conn)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 115, in wrapper
    return func(*args,**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 294, in get_track_id
    artist_ids = [get_artist_id(a,dbconn=dbconn) for a in trackdict['artists']]
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 294, in <listcomp>
    artist_ids = [get_artist_id(a,dbconn=dbconn) for a in trackdict['artists']]
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/dbcache.py", line 49, in outer_func
    result = inner_func(*args,**kwargs,dbconn=conn)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 115, in wrapper
    return func(*args,**kwargs)
  File "/usr/local/lib/python3.9/dist-packages/maloja/database/sqldb.py", line 346, in get_artist_id
    op = DB['artists'].select(
TypeError: select() takes 1 positional argument but 2 were given
```

**Comment by bbsixzz** [comment]
I installed the docker image and everything works fine.

The problem seems localized to `pip install malojaserver` on debian.

**Comment by krateng** [comment]
Looks like this is also due to compatibility with SQLAlchemy 2.0, which should be fixed by 3ae395f
